src/main/java/org/graylog2/gelfclient/transport/GelfHttpTransport.java

```diff
@@ -0,0 +1,135 @@
+        senderThreadReference.set(senderThread);
+
+        bootstrap.group(workerGroup)
+                .channel(NioSocketChannel.class)
+                .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, config.getConnectTimeout())
+                .option(ChannelOption.TCP_NODELAY, config.isTcpNoDelay())
+                .option(ChannelOption.SO_KEEPALIVE, config.isTcpKeepAlive())
+                .remoteAddress(config.getRemoteAddress())
+                .handler(new ChannelInitializer<SocketChannel>() {
+                    @Override
+                    protected void initChannel(SocketChannel ch) throws Exception {
+                        if (config.isTlsEnabled()) {
+                            LOG.debug("TLS enabled.");
+                            final SslContext sslContext;
+
+                            if (!config.isTlsCertVerificationEnabled()) {
+                                // If the cert should not be verified just use an insecure trust manager.
+                                LOG.debug("TLS certificate verification disabled!");
+                                sslContext = SslContextBuilder.forClient()
+                                        .trustManager(InsecureTrustManagerFactory.INSTANCE)
+                                        .build();
+                            } else if (config.getTlsTrustCertChainFile() != null) {
+                                // If a cert chain file is set, use it.
+                                LOG.debug("TLS certificate chain file: {}", config.getTlsTrustCertChainFile());
+                                sslContext = SslContextBuilder.forClient()
+                                        .trustManager(config.getTlsTrustCertChainFile())
+                                        .build();
+                            } else {
+                                // Otherwise use the JVM default cert chain.
+                                sslContext = SslContextBuilder.forClient().build();
+                            }
+
+                            ch.pipeline().addLast(sslContext.newHandler(ch.alloc()));
+                        }
+
+                        ch.pipeline().addLast(new HttpClientCodec());
+                        ch.pipeline().addLast(new HttpContentDecompressor());
+                        ch.pipeline().addLast(new GelfHttpEncoder(config.getUri()));
+                        ch.pipeline().addLast(new GelfMessageJsonEncoder());
+                        ch.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                            @Override
+                            public void channelActive(ChannelHandlerContext ctx) throws Exception {
+                                senderThread.start(ctx.channel());
+                            }
+
+                            @Override
+                            public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+                                LOG.info("Channel disconnected!");
```

Comment:
I see lots of these errors when running against a real HTTP server and it also drops messages. 

```
[gelfHttpTransport-1-1] INFO org.graylog2.gelfclient.transport.GelfHttpTransport - Channel disconnected!
[gelfHttpTransport-1-1] ERROR org.graylog2.gelfclient.encoder.GelfHttpEncoder - Error while encoding HTTP request
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:247)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1147)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:347)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:700)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:635)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:552)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:514)
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
[gelfHttpTransport-1-1] INFO org.graylog2.gelfclient.transport.GelfHttpTransport - Channel disconnected!
[gelfHttpTransport-1-1] INFO org.graylog2.gelfclient.transport.GelfHttpTransport - Channel disconnected!
[gelfHttpTransport-1-1] ERROR org.graylog2.gelfclient.encoder.GelfHttpEncoder - Error while encoding HTTP request
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:247)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1147)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:347)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:700)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:635)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:552)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:514)
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
[gelfHttpTransport-1-1] INFO org.graylog2.gelfclient.transport.GelfHttpTransport - Channel disconnected!
[gelfHttpTransport-1-1] INFO org.graylog2.gelfclient.transport.GelfHttpTransport - Channel disconnected!
```